pip_audit/_dependency_source/requirement.py

```diff
@@ -84,10 +84,40 @@ def collect(self) -> Iterator[Dependency]:
         Raises a `RequirementSourceError` on any errors.
         """
 
+        with ExitStack() as stack:
+            # We need to handle process substitution inputs so we can invoke
+            # `pip-audit` like so:
+            #
+            #   pip-audit -r <(echo 'something')
+            #
+            # Since `/dev/fd/<n>` inputs are unique to the parent process, we
+            # can't pass these file names to `pip` and expect `pip` to able to
+            # read them.
+            #
+            # In order to get around this, we're going to copy each input into a
+            # a corresponding temporary file and then pass that set of files
+            # into `pip`.
+            tmp_files: list[IO[str]] = [
+                stack.enter_context(NamedTemporaryFile(mode="w")) for _ in self._filenames
+            ]
+
+            # For each input file, copy it to one of our temporary files.
+            # Ensure we flush and fsync so our writes are visible to `pip`.
+            for filename, tmp_file in zip(self._filenames, tmp_files):
```

Comment:
Minor thing, but rather than two loops and a `zip`, maybe we can just do a single loop here? Something like:

```python
tmp_files = []
for filename in self._filenames:
    tmp_file = stack.enter_context(...)
    with filename.open("r") as io:
        ...
        
# use tmp_files here
```

Reply:
Yep, that works.

Reply:
[b14e1b5](https://github.com/pypa/pip-audit/pull/538/commits/b14e1b55942b606db97efc60ae2dc31f8cb63bf0)